lib/src/model/game/chat_presets_controller.dart

```diff
@@ -0,0 +1,107 @@
+import 'package:freezed_annotation/freezed_annotation.dart';
+import 'package:lichess_mobile/src/model/common/id.dart';
+import 'package:lichess_mobile/src/model/game/chat_controller.dart';
+import 'package:lichess_mobile/src/model/game/game_controller.dart';
+import 'package:lichess_mobile/src/model/game/message_presets.dart';
+import 'package:riverpod_annotation/riverpod_annotation.dart';
+
+part 'chat_presets_controller.freezed.dart';
+part 'chat_presets_controller.g.dart';
+
+@riverpod
+class ChatPresetsController extends _$ChatPresetsController {
+  late GameFullId _gameId;
+
+  static const Map<PresetMessageGroup, List<PresetMessage>> _presetMessages = {
+    PresetMessageGroup.start: [
+      (label: 'HI', value: 'Hello'),
+      (label: 'GL', value: 'Good luck'),
+      (label: 'HF', value: 'Have fun!'),
+      (label: 'U2', value: 'You too!'),
+    ],
+    PresetMessageGroup.end: [
+      (label: 'GG', value: 'Good game'),
+      (label: 'WP', value: 'Well played'),
+      (label: 'TY', value: 'Thank you'),
+      (label: 'GTG', value: "I've got to go"),
+      (label: 'BYE', value: 'Bye!'),
+    ],
+  };
+
+  @override
+  Future<ChatPresetsState> build(GameFullId id) async {
+    _gameId = id;
+
+    final gameState = ref.read(gameControllerProvider(_gameId)).value;
+
+    ref.listen(gameControllerProvider(_gameId), _handleGameStateChange);
```

Comment:
Here I don't think `ref.listen` is the best way to handle this. 

Since we want to act upon the game state, `ref.watch(gameControllerProvider(id))` feels more like what we want. 
We don't want to rebuild the chat state each time the game state changes, so you can also use [selectAsync](https://riverpod.dev/docs/advanced/select#selecting-asynchronous-properties) to filter rebuild by returning only `PresetMessageGroup` when they change.